marconi-chain-index/test/CompareToDbSync.hs

```diff
@@ -0,0 +1,200 @@
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE LambdaCase         #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# OPTIONS_GHC -Wno-orphans    #-}
+module CompareToDbSync where
+
+import Control.Lens ((^.))
+import Control.Monad (forM_)
+import Control.Monad.IO.Class (liftIO)
+import Control.Monad.Trans.Except (runExceptT)
+import Data.ByteString qualified as BS
+import Data.Coerce (coerce)
+import Data.Map.Strict qualified as Map
+import Data.Ratio (denominator, numerator)
+import Data.Word (Word64)
+import Database.PostgreSQL.Simple qualified as PG
+import Database.PostgreSQL.Simple.FromField qualified as PG
+import Database.PostgreSQL.Simple.ToField qualified as PG
+import System.Environment (lookupEnv)
+import System.FilePath ((</>))
+
+import Cardano.Api qualified as C
+import Cardano.Api.Shelley qualified as C
+import Cardano.Crypto.Hash qualified as Crypto
+import Cardano.Ledger.Shelley.API qualified as Ledger
+import Ouroboros.Consensus.Cardano.Block qualified as O
+import Ouroboros.Consensus.Config qualified as O
+import Ouroboros.Consensus.Node qualified as O
+
+import Marconi.ChainIndex.Indexers.EpochState qualified as EpochState
+import Marconi.ChainIndex.Node.Client.GenesisConfig qualified as GenesisConfig
+import Marconi.ChainIndex.Types (epochStateDbName)
+import Marconi.ChainIndex.Utils qualified as Utils
+import Marconi.Core.Storable qualified as Storable
+
+import Hedgehog ((===))
+import Hedgehog qualified as H
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Range qualified as Range
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Hedgehog (testPropertyNamed)
+
+
+tests :: TestTree
+tests = testGroup "Marconi to cardano-db-sync comparisons"
+  [ testPropertyNamed
+      "Compare *all* epoch nonces between Marconi and cardano-db-sync"
+      "propEpochNonce" propEpochNonce
+  , testPropertyNamed
+      "Compare random 10 epoch stakepool sizes between Marconi and cardano-db-sync"
```

Comment:
Why 10 random? It's not better to compare **all** SDD for all epochs?

Reply:
All is better yes! It only takes a few minutes. Initially I thought it took much longer, but I was doing the "compare all" a hundred times (the hedgehog default number of tests)